packages/TorneloScoresheet/src/components/TextIconButton/TextIconButton.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import {
+  StyleProp,
+  TextStyle,
+  TouchableOpacity,
+  TouchableOpacityProps,
+} from 'react-native';
+import { SvgProps } from 'react-native-svg';
+import { colours } from '../../style/colour';
+import PrimaryText from '../PrimaryText/PrimaryText';
+import RoundedView from '../RoundedView/RoundedView';
+import { styles } from './style';
+
+type TextIconButtonParms = {
+  Icon?: React.FC<SvgProps>;
```

Comment:
This is an unusual API design. I think it might be more appropriate to accept the icon as a string and render an 'react-native-vector-icons/MaterialIcons' `Icon` accordingly? (See our `IconButton` component). Curious to hear your thoughts, because it probably doesn't matter massively either way, just seems unnecessarily generic/abstract or smt?

Reply:
This resticts us to Google Font Material icons though, I'm using the icons straight from the figma file, those can be downloaded in svg form. This is the same approach used for our `PieceAssets` as well as the `ActionButton` components, I think its fine this way as SVG images are superior to all others, and this way we have the freedom to use any icon

Reply:
FYI, the main Tornelo system currently has the same policy with downloading svg.